### PR TITLE
Reviewing operators in Postgres.

### PR DESCRIPTION
**Pull request by bernardoamc**
Podem me ajudar a revisar? @laurocaetano @erichkist

### CODE REVIEW

File: _posts/2015-07-23-postgres-review-operators.md [hunk 1]
```diff
@@ -0,0 +1,292 @@
+---
+layout:     post
+title:      Review of operators in Postgres
+date:       2015-07-23 23:00:00
+summary:    In this post we will review some operators so we can have
```
Comment (erichkist) [inline _posts/2015-07-23-postgres-review-operators.md]: If this summary is showed in someplace I believe it needs change. The user doesn't know the next post, so the post needs to be unique. What will I learn?

File: _posts/2015-07-23-postgres-review-operators.md [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+            a place to reference in future posts.
+categories: sql
+---
+
+Let's investigate some operators in Postgres and how to use them
+in our day-to-day work. The objective is to have a centralized guide
```
Comment (erichkist) [inline _posts/2015-07-23-postgres-review-operators.md]: `The objective is to have a centralized guide.`

File: _posts/2015-07-23-postgres-review-operators.md [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+
+SELECT array[1,3,6] @> array[1,3];
+
+ ?column?
+----------
+ t
+(1 row)
+{% endhighlight %}
+
++ `<@` on the other hand specifies if an Array **is contained by** another one.
```
Comment (erichkist) [inline _posts/2015-07-23-postgres-review-operators.md]: `<@` specifies if an Array **is contained by** another one.

File: _posts/2015-07-23-postgres-review-operators.md [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+
+{% highlight sql %}
+SELECT array[3,6] && array[1,4,7];
+
+ ?column?
+----------
+ f
+(1 row)
+
+SELECT array[3,6] && array[1,3,6];
```
Comment (erichkist) [inline _posts/2015-07-23-postgres-review-operators.md]: I would use `SELECT array[2,3,6] && array[1,3,6];` just to explicit the answer will be `true` if I use an element (number 2) that isn't there but the others are (number 3 and 6).

File: _posts/2015-07-23-postgres-review-operators.md [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+(1 row)
+{% endhighlight %}
+
+*We cannot chain `->>` since it returns a TEXT.*
+
++ `#>` and `#>>` receives a path as an Array and returns a JSON and
+TEXT respectively. Both return `NULL` if the path cannot be found.
+
+{% highlight sql %}
+SELECT '{"state": {"name": "abc"} }'::json#>ARRAY['country', 'name'];
```
Comment (erichkist) [inline _posts/2015-07-23-postgres-review-operators.md]: I think the array params is wrong (`s/country/state/`) :

``` sql
amgen-forms_development=# SELECT '{"state": {"name": "abc"} }'::json#>ARRAY['country', 'name'];
 ?column?
----------

(1 row)

amgen-forms_development=# SELECT '{"state": {"name": "abc"} }'::json#>ARRAY['state', 'name'];
 ?column?
----------
 "abc"
(1 row)
```

File: _posts/2015-07-23-postgres-review-operators.md [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+ text
+(1 row)
+{% endhighlight %}
+
+### JSONB
+
+`JSONB` adds new operators on top of the existing ones for the `JSON` datatype.
+
++ `@>` and `<@` operators performs like the ones we saw for the Array datatype.
+The first checks if the left JSONB contains the one in the right and the latter
```
Comment (erichkist) [inline _posts/2015-07-23-postgres-review-operators.md]: I will love if I had the same explanation (maybe partial or another) in the Array operator topic: who is compared with what.

**Comment by erichkist** [comment]
:heart: :heart: :heart: :heart: :heart: :heart: 

I loved the post. I learned a lot with it. Thank you :)

File: _posts/2015-07-23-postgres-review-operators.md (hunk 5 above)
Comment (bernardoamc) [inline _posts/2015-07-23-postgres-review-operators.md]: :runner:

File: _posts/2015-07-23-postgres-review-operators.md (hunk 4 above)
Comment (bernardoamc) [inline _posts/2015-07-23-postgres-review-operators.md]: :+1:

File: _posts/2015-07-23-postgres-review-operators.md (hunk 3 above)
Comment (bernardoamc) [inline _posts/2015-07-23-postgres-review-operators.md]: :thumbsup:

File: _posts/2015-07-23-postgres-review-operators.md (hunk 1 above)
Comment (bernardoamc) [inline _posts/2015-07-23-postgres-review-operators.md]: Fair point, this is the short description that is placed below the post title in the posts list.

File: _posts/2015-07-23-postgres-review-operators.md (hunk 6 above)
Comment (bernardoamc) [inline _posts/2015-07-23-postgres-review-operators.md]: Do your prefer this description over the one I did on the `Array` section?

Comment (erichkist) [inline _posts/2015-07-23-postgres-review-operators.md]: I don't know. Maybe just explicit the order of compare.

Comment (bernardoamc) [inline _posts/2015-07-23-postgres-review-operators.md]: :+1:

**Pull request merged**